### ISSUE TITLE
ChanUpgradeAck should call the chanUpgradeOpen subprotocol if ready

### ISSUE THREAD

**Issue by damiannolan** [cosmos/ibc-go#4030]
## Summary

The following spec additions should be added to the `ChanUpgradeAck` handler to call the channel upgrade open subprotocol if ready:

```
// if both sides have already flushed then open the upgrade handshake immediately
if currentChannel.flushStatus == FLUSHCOMPLETE && counterpartyFlushStatus == FLUSHCOMPLETE {
        openUpgradelHandshake(portIdentifier, channelIdentifier)
        module.onChanUpgradeOpen(portIdentifier, channelIdentifier)
}
```

Note, we normally route app callbacks from the core msg server so we probably need to make this check at that level in order to call:

```go
cbs.OnChanUpgradeOpen(ctx, msg.PortId, msg.ChannelId)
```
____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by damiannolan** [comment]
Note when calling into `ChanUpgradeOpen` subprotocol from `ChanUpgradeAck` we assume the `counterpartyChannelState` to be `TRYUPGRADE` as is already verified in the `ChanUpgradeAck` handler
```go
func (k Keeper) ChanUpgradeOpen(
	ctx sdk.Context,
	portID,
	channelID string,
	counterpartyChannelState types.State,
	proofChannel []byte,
	proofHeight clienttypes.Height,
) error 
```

**Comment by DimitrisJim** [comment]
closed in #4075
